src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3444 @@
+            ret = 0;
+            if (EnAn_ChangeCutscene(this, csId)) {
+                goto label;
+            }
+            break;
+
+        case 0x2:
+        case 0x4:
+        case 0x6:
+        case 0x8:
+        label:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)),
+                                      this->actor.child);
+            }
+            ret = 1;
+            this->msgEventState++;
+            break;
+
+        case 0x1:
+        case 0x3:
+        case 0x5:
+        case 0x7:
+            Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)), &this->actor);
+            ret = 1;
+            this->msgEventState++;
+            break;
+
+        case 0x9:
+            CutsceneManager_Stop(csId);
+            ret = 1;
+            this->msgEventState++;
+            break;
+
+        default:
+            break;
+    }
+
+    return ret;
+}
+
+// Only used if Player is using Kafei's Mask or if Human and Promised midnight meeting
+s32 EnAn_MsgEvent_MidnightMeeting(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+
+    if (this->msgEventState == 0) {
+        func_800B7298(play, &this->actor, PLAYER_CSACTION_WAIT);
+        play->nextEntrance = ENTRANCE(STOCK_POT_INN, 3);
+        gSaveContext.nextCutsceneIndex = 0;
+        play->transitionTrigger = TRANS_TRIGGER_START;
+        play->transitionType = TRANS_TYPE_FADE_BLACK;
+        gSaveContext.nextTransitionType = TRANS_TYPE_FADE_BLACK_SLOW;
+        this->msgEventState++;
+    }
+
+    return 0;
+}
+
+s32 EnAn_MsgEvent_Cooking(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+    s32 ret = false;
+
+    switch (this->msgEventState) {
+        case 0x0:
+            EnAn_ChangeAnim(this, play, ENAN_ANIM_TASTE);
+            this->msgEventState++;
+            break;
+
+        case 0x1:
+        case 0x3:
+            if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+                this->msgEventState++;
+                ret = true;
+            }
+            break;
+
+        case 0x2:
+            EnAn_ChangeAnim(this, play, ENAN_ANIM_TASTE_END);
+            this->msgEventState++;
+            break;
+
+        default:
+            break;
+    }
+
+    return ret;
+}
+
+s32 EnAn_MsgEvent_LaundryPool(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+    s32 ret = false;
+
+    switch (this->msgEventState) {
+        case 0x0:
+            if ((Player_GetMask(play) == PLAYER_MASK_KAFEIS_MASK) ||
+                CHECK_WEEKEVENTREG(WEEKEVENTREG_SPOKE_TO_ANJU_IN_LAUNDRY_POOL)) {
+                this->msgEventState++;
+                goto label;
+            } else {
+                ret = true;
+                this->stateFlags |= ENAN_STATE_DRAW_KAFEI_MASK;
+                this->msgEventState++;
+            }
+            break;
+
```

Comment:
As mentioned you could just put the break in the else and just fall through instead of the goto. However i'll admit, not sure that is better in this particular sitation.

Reply:
I'm not sure either.
I'm not sure if being consistent with this kind of thing (at least on the same actor) is worth either

Reply:
What about something like this

```c
switch (this->msgEventState) {
        case 0x0:
            if ((Player_GetMask(play) == PLAYER_MASK_KAFEIS_MASK) ||
                CHECK_WEEKEVENTREG(WEEKEVENTREG_SPOKE_TO_ANJU_IN_LAUNDRY_POOL)) {
                this->msgEventState++;
                // fallthrough
            } else {
                ret = true;
                this->stateFlags |= ENAN_STATE_DRAW_KAFEI_MASK;
                this->msgEventState++;
                break;
            }
            FALLTHROUGH
 ```
Just to make it clear the `if` block falls through while the `else` block does not.

Reply:
I'm not completely convinced by this, it looks a bit ugly.
In the meanwhile I changed the others to not use a goto

Reply:
I think I prefer the one without a `goto` here